google/ground-android
# Download and render offline areas

This PR includes a couple of changes related to downloading and rendering offline areas:

- Add an offlineAreaRepository
- Download areas on download button click in the OfflineAreaSelector Fragment
- Render OfflineArea tiles using the MapBox Tile Provider

To verify these changes:

- Start the app
- select offline areas from the main menu
- Select the + icon to add a new area
- zoom in to a portion of the map then click download.
- return to the home screen, after the tiles finish downloading, areas will render on the map. Select Normal mode to easily distinguish downloaded imagery from the default map -- imagery will only render at zoom level 7.
- You can also check the device storage in android studio to verify tiles have been downloaded to the device.

Fixes #416
